main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -767,6 +767,9 @@ object Lowering {
       val defExp = LoweredAst.Expression.Def(sym, defTpe, loc)
       val argExps = mkPredSym(pred) :: visitExp(exp) :: Nil
       LoweredAst.Expression.Apply(defExp, argExps, tpe, pur, eff, loc)
+
+    case TypedAst.Expression.Error(m, _, _, _) => ???
```

Comment:
Lets make this into a HoleError or alternatively an InternalCompilerException.

Reply:
I'm not seeing any `HoleError`s for LoweredAst

Reply:
Ah- its just called `Hole`. But actually I prefer the use of the exception.